File: src/main/java/org/apache/prepbuddy/examples/FacetsMain.java
package org.apache.prepbuddy.examples;

import org.apache.prepbuddy.groupingops.TextFacets;
import org.apache.prepbuddy.rdds.TransformableRDD;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

public class FacetsMain {
    public static void main(String[] args) {
        if (args.length == 0) {
            System.out.println("--> File Path Need To Be Specified");
            System.exit(0);
        }
        SparkConf conf = new SparkConf().setAppName("Facets");
        JavaSparkContext sc = new JavaSparkContext(conf);

        String filePath = args[0];
        JavaRDD<String> csvInput = sc.textFile(filePath, Integer.parseInt(args[1]));

        TransformableRDD inputRdd = new TransformableRDD(csvInput);
        TextFacets textFacets = inputRdd.listFacets(4);

        long count = textFacets.count();
        System.out.println("-->>> Total " + count);

        sc.close();
    }
}

File: src/main/scala/org/apache/datacommons/prepbuddy/examples/facetsMain.scala
package org.apache.datacommons.prepbuddy.examples

import org.apache.datacommons.prepbuddy.clusterers.TextFacets
import org.apache.datacommons.prepbuddy.rdds.TransformableRDD
import org.apache.spark.rdd.RDD
import org.apache.spark.{SparkConf, SparkContext}


object FacetsMain {
    def main(args: Array[String]) {
        if (args.length == 0) {
            System.out.println("--> File Path Need To Be Specified")
            System.exit(0)
        }
        val conf: SparkConf = new SparkConf().setAppName("Facets")
        val sc: SparkContext = new SparkContext(conf)
        val filePath: String = args(0)
        val csvInput: RDD[String] = sc.textFile(filePath, args(1).toInt)
        val inputRdd: TransformableRDD = new TransformableRDD(csvInput)
        val textFacets: TextFacets = inputRdd.listFacets(1)
        val count: Long = textFacets.count
        System.out.println("-->>> Total " + count)
        sc.stop()
    }
}

